### DISCUSSION TITLE
New project npx next build error

### DISCUSSION THREAD

**Discussion by RobertKoval** [vercel/next.js#72986] [Help]
### Link to the code that reproduces this issue

https://github.com/RobertKoval/issue_nextjs

### To Reproduce

1) Create a new project using `npx create-next-app@latest`
2) run `npx next build `

### Current vs. Expected behavior

I expected the project to compile, but instead, I encountered an error:

```
(node:9416) ExperimentalWarning: CommonJS module /opt/homebrew/lib/node_modules/npm/node_modules/debug/src/node.js is loading ES Module /opt/homebrew/lib/node_modules/npm/node_modules/supports-color/index.js using require().
Support for loading ES Module in require() is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
   ▲ Next.js 15.0.3

   Creating an optimized production build ...
Failed to compile.

./app/page.tsx + 1 modules
Unexpected end of JSON input


> Build failed because of webpack errors
```

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.5.0: Wed May  1 20:12:58 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6000
  Available memory (MB): 16384
  Available CPU cores: 10
Binaries:
  Node: 23.2.0
  npm: 10.9.0
  Yarn: 1.22.22
  pnpm: 9.4.0
Relevant Packages:
  next: 15.0.3 // Latest available version is detected (15.0.3).
  eslint-config-next: 15.0.3
  react: 19.0.0-rc-66855b96-20241106
  react-dom: 19.0.0-rc-66855b96-20241106
  typescript: 5.6.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

create-next-app, Developer Experience, Webpack

### Which stage(s) are affected? (Select all that apply)

next build (local)

**Comment by RobertKoval** [comment]
I tried with different NodeJS versions, **v21.7.3** and **v22.11.0** works fine.

**Comment by JulioBarros** [comment]
I have this same issue with a brand new project with no edits just `npx create-next-app@latest  buildtest` and then `npm run build` fails with 

```
❯ npm run build

> buildtest@0.1.0 build
> next build

   ▲ Next.js 15.0.3

   Creating an optimized production build ...
Failed to compile.

./src/app/page.tsx + 1 modules
Unexpected end of JSON input

> Build failed because of webpack errors
```

I'm on node version v23.2.0

**Comment by jmgibson1976** [comment]
```
> next build

   ▲ Next.js 15.0.3

   Creating an optimized production build ...
Failed to compile.

./app/page.tsx + 1 modules
Unexpected end of JSON input


> Build failed because of webpack errors
```

Also on node version v23.2.0

**Comment by jmgibson1976** [comment]
@RobertKoval @JulioBarros 

Remove the → from the text in the last link (Go to nextjs.org → becomes Go to nextjs.org) and it seems to compile fine.

Possibly improper escaping of unicode character.

I created a const:
  const goto = "Go to nextjs.org \u2192";
  
 and then replaced the line: Go to nextjs.org →
 
 with: {goto}

and it built as well.

Take what I say with a grain of salt, im very new to typescript, react, and nextjs.

It only happens on build so perhaps there is a problem with a minimizer or other tool that webpack is configured to use?!

**Comment by JulioBarros** [comment]
That is true.  It does work if you remove the "→" from the default page. So that is a good hint

I'm not using the that char or any other that might need encoding on my page though.  I am using zod and react-hook-form and the only JSON on it is a valid zod spec.  If I remove the form and zod spec it builds and it all builds fine in dev mode.

Right now I'm getting
~~~

   Creating an optimized production build ...
Failed to compile.

./src/app/contact/page.tsx + 4 modules
Unexpected end of JSON input

./node_modules/remark-html/lib/index.js + 80 modules
Unexpected end of JSON input

./node_modules/remark/index.js + 110 modules
Unexpected end of JSON input

./node_modules/unist-util-visit-parents/lib/index.js + 1 modules
Unexpected end of JSON input
~~~

**Comment by agoe** [comment]
maybe  this helps i have nextjs 14 latest and the same problems with the build.
Found this:

23.2 breaks a few packages that use json trough webpack (date-fns) #55826
https://github.com/nodejs/node/issues/55826

downgraded my node installation from 23.2 to 20.18 et voila

**Reply by agoe** [reply]
Overlooked that [RobertKoval] also downgraded his node version and this solved the issue. Will stop upgrading my node version without proper testing.

**Comment by davidar** [comment]
I'm seeing the same issue on node 23.2, no issues on node 23.1

_Edit by maintainer bot: Comment was **automatically** minimized because it was considered unhelpful. (If you think this was by mistake, let us know). Please only comment if it adds context to the issue. If you want to express that you have the same problem, use the upvote 👍 on the issue description or subscribe to the issue for updates. Thanks!_

**Comment by JulioBarros** [comment]
Ok, so I use brew and did an upgrade and it installed a new version of node v23.2.0 (same version but now v23.2.0_1 when it installed. I see this commit that changes node https://github.com/Homebrew/homebrew-core/commit/a6cd670526b0b1b7abe7e2610fb55536ea712280) 

Anyway, now it gets further except I have an eslint issue. I'm going to assume that is my fault for now, though I can't figure out why and it works fine in dev.  The error I have now is below in case anybody has any ideas.

```
Type error: Type '{ params: { slug: string; }; }' does not satisfy the constraint 'PageProps'.
  Types of property 'params' are incompatible.
    Type '{ slug: string; }' is missing the following properties from type 'Promise<any>': then, catch, finally, [Symbol.toStringTag]
```

**Comment by JorgeMadson** [comment]
I using windows and got the same error on node 23, downgrading to node 20 solved the issue.
The error:
```   Creating an optimized production build ...
Failed to compile.

./node_modules/next/dist/build/webpack/loaders/next-app-loader/index.js?page=%2Fapi%2Fimages%2Fupload%2Froute&name=app%2Fapi%2Fimages%2Fupload%2Froute&pagePath=private-next-app-dir%2Fapi%2Fimages%2Fupload%2Froute.ts&appDir=C%3A%5CUsers%5Cfilho%5Cdev%5Cpalestra-next%5Csrc%5Capp&appPaths=%2Fapi%2Fimages%2Fupload%2Froute&pageExtensions=tsx&pageExtensions=ts&pageExtensions=jsx&pageExtensions=js&basePath=&assetPrefix=&nextConfigOutput=&flyingShuttle=false&nextConfigExperimentalUseEarlyImport=&preferredRegion=&middlewareConfig=e30%3D! + 4 modules
Unexpected end of JSON input
```

**Comment by JulioBarros** [comment]
can also confirm v20.18.0 does not have the issue.

**Comment by js** [comment]
I noticed this in one of my Docker images as well, had to pin it to the `node:23.1-alpine3.19` image rather than latest

**Comment by zed76r** [comment]
It appears to be resolved in the 23.3.x version of Node.js.

**Comment by adriensoulie** [comment]
I had this issue and after downgrading Node it was fine and the build passed

**Comment by SheIsBukki** [comment]
I'm currently facing similar issue but with NextJS 15

**When I ran npm run build on local, it threw these errors**
➜  ecommerce-app git:(main) npm run build

> ecommerce-app@0.1.0 build
> next build

   ▲ Next.js 15.1.4
   - Environments: .env

   Creating an optimized production build ...
Failed to compile.

./node_modules/date-fns/esm/format/index.js + 4 modules
Unexpected end of JSON input

./node_modules/xstate/dist/raise-c17ec2bc.esm.js + 1 modules
Unexpected end of JSON input

./sanity.config.ts + 8 modules
Unexpected end of JSON input

./src/components/cart/Cart.tsx + 1 modules
Unexpected end of JSON input

./src/app/product/[id]/page.tsx + 7 modules
Unexpected end of JSON input


> Build failed because of webpack errors
➜  ecommerce-app git:(main) 


**But this is the error it's throwing on Vercel**
Failed to compile.
./src/actions/auth.ts:8:15
Type error: Module '"@prisma/client"' has no exported member 'User'.
   6 | } from "@oslojs/encoding";
   7 | import { sha256 } from "@oslojs/crypto/sha2";
>  8 | import type { User, Session } from "@prisma/client";
     |               ^
   9 | import { cookies } from "next/headers";
  10 | import { cache } from "react";
  11 | import prisma from "@/lib/prisma";
Static worker exited with code: 1 and signal: null
Error: Command "npm run build" exited with 1


Please note that the line it is referencing in the deployment error is actually not throwing any errors, not even a squiggly line in the project in the editor. And it is working perfectly, the database reflects the User model. Prisma studio is showing users, I am also testing with three different test email addresses and they all work. I don't understand because the error it is throwing is precise but basically meaningless because there is an exported member 'User'

removing 'type' from the import to use this approach doesn't resolve the deployment error, but it also doesn't throw any errors in the project or the development version and doesn't disconnect the database or whatever

Changing the import to this: import { Session, User } from "@prisma/client"; throws the Vercel deployment error below

Failed to compile.
./src/actions/auth.ts:12:10
Type error: Module '"@prisma/client"' has no exported member 'Session'.
  10 | import { cache } from "react";
  11 | import prisma from "@/lib/prisma";
> 12 | import { Session, User } from "@prisma/client";
     |          ^
  13 |
  14 | /* SESSION */
  15 |
Static worker exited with code: 1 and signal: null
Error: Command "npm run build" exited with 1


The dependencies and devDependencies in the project

```json
"dependencies": {
    "@oslojs/crypto": "^1.0.1",
    "@oslojs/encoding": "^1.1.0",
    "@prisma/client": "^6.2.1",
    "@sanity/image-url": "^1.1.0",
    "@sanity/vision": "^3.70.0",
    "dayjs": "^1.11.13",
    "lucide-react": "^0.473.0",
    "moment": "^2.30.1",
    "next": "15.1.4",
    "next-sanity": "^9.8.38",
    "react": "^19.0.0",
    "react-dom": "^19.0.0",
    "react-icons": "^5.4.0",
    "sanity": "^3.70.0",
    "stripe": "^17.5.0",
    "styled-components": "^6.1.14",
    "zod": "^3.24.1",
    "zustand": "^5.0.3"
  },
  "devDependencies": {
    "@eslint/eslintrc": "^3",
    "@types/node": "^20",
    "@types/react": "^19",
    "@types/react-dom": "^19",
    "eslint": "^9",
    "eslint-config-next": "15.1.4",
    "eslint-plugin-no-relative-import-paths": "^1.6.1",
    "postcss": "^8",
    "prettier": "^3.4.2",
    "prettier-plugin-tailwindcss": "^0.6.10",
    "prisma": "^6.2.1",
    "tailwindcss": "^3.4.1",
    "typescript": "^5"
  }
```
